packages/eslint-plugin/src/rules/sort-type-constituents.ts

```diff
@@ -170,6 +170,41 @@ export default createRule<Options, MessageIds>({
       numeric: true,
     });
 
+    // Normalize source text for complex types to ignore formatting differences.
+    function getSortText(
+      type: TSESTree.TypeNode | TSESTree.TypeElement,
+    ): string {
```

Comment:
[Bug] As convenient as `sourceCode.getText` is, most cases where it's used are susceptible to bugs. Here's an example for type parameters in mapped types:

```ts
type _ = {
  [P in 'a' /* now gets included */ | 'b']: string
}
```

This function will need to account for all the AST types that might have comments / formatting differences, included `TSUnionType`, `TSIntersectionType`, and `TSTypePredicate`.